### PR TITLE
Virtual Network Gateway new Properties in BgpSettings

### PR DESCRIPTION
**Pull request by Khushboo-Baheti**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [X] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [X] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [X] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by akning-ms** [comment]
re-assign to @lirenhe for networking PR review

**Comment by Khushboo-Baheti** [comment]
NRP release for these new properties in BgpSettings are on hold because of some regression. please contact to me before merging this PR.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json]: This could be considered a breaking change in SDKs

File: specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json]:
```suggestion
          "description": "BGP peering address with IP configuration ID for virtual network gateway."
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json]:
```suggestion
          "description": "The ID of IP configuration which belongs to gateway."
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json]:
```suggestion
          "description": "The list of default BGP peering addresses which belong to IP configuration."
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json]:
```suggestion
          "description": "The list of custom BGP peering addresses which belong to IP configuration."
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json]:
```suggestion
          "description": "The list of tunnel public IP addresses which belong to IP configuration."
```

**Review by anton-evseev — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json
Comment (Khushboo-Baheti) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json]: Sure I will make these changes.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json
Comment (Khushboo-Baheti) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json]: We are discussing about it. I will give update soon. Thanks @number213

Comment (Khushboo-Baheti) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json]: I have removed this breaking change and added properties in existing BgpSettings.

Comment (Khushboo-Baheti) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json]: Fixed this comment so resolving.

**Comment by ravbhatnagar** [comment]
This api version hasn't been merged to master. And hence this change will not cause a breaking change. Signing off from ARM side.

**Review by anton-evseev — APPROVED** [review]

**Comment by number213** [comment]
@lirenhe approve from both ARM & NRP's sides. Could you please merge?

**Review by lirenhe — APPROVED** [review]

**Pull request merged**
